VideoLocker/src/main/java/org/edx/mobile/http/RetroHttpException.java

```diff
@@ -1,13 +1,24 @@
 package org.edx.mobile.http;
 
+import android.support.annotation.NonNull;
+
 import retrofit.RetrofitError;
 
 /**
- * This sould be the base class of all http exception?
+ * Should this be the base class of all http exceptions?
  */
-public class RetroHttpException extends Exception{
-    public final RetrofitError cause;
-    public RetroHttpException(RetrofitError cause){
+public class RetroHttpException extends Exception {
```

Comment:
I assume the purpose of this class is to use checked exceptions instead of the unchecked `RetrofitError`? In that case it might be more convenient to make it a complete wrapper, implementing all the methods in `RetrofitError`, and calling through to it.